src/ts/spatialnavigation/navigationalgorithm.ts

```diff
@@ -59,9 +59,20 @@ function distance(a: Vector, b: Vector): number {
 function getElementVector(element: HTMLElement): Vector {
   const boundingRect = element.getBoundingClientRect();
 
+  let _x: number;
+  let _y: number;
+
+  if (boundingRect.hasOwnProperty('x') && boundingRect.hasOwnProperty('y')) {
```

Comment:
I'd suggest we use
```suggestion
  if (typeof boundingRect.x === 'number' && typeof boundingRect.y === 'number') {
```
instead of `hasOwnProperty` to prevent any issues with inheritance.

Reply:
Makes sense - especially considering jsdom has not fully implemented the DOMRect class. Updated in latest commit